x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/EsqlFeatures.java

```diff
@@ -121,6 +121,11 @@ public class EsqlFeatures implements FeatureSpecification {
      */
     public static final NodeFeature MV_ORDERING_SORTED_ASCENDING = new NodeFeature("esql.mv_ordering_sorted_ascending");
 
+    /**
+     * Cast string literals to a desired data type for IN predicate and more types for BinaryComparison.
+     */
+    public static final NodeFeature STRING_LITERAL_AUTO_CASTING_815 = new NodeFeature("esql.string_literal_auto_casting_8_15_0");
```

Comment:
I'm not very sure if I should add a new entry in the EsqlFeatures, it is the extension of a previous entry, but in different release.